Core/src/Material/Interactions.cpp

```diff
@@ -97,36 +75,63 @@ inline float logDeriveWMax(float mass, float qOverP,
 /// where (Z/A)*rho is the electron density in the material and x is the
 /// traversed length (thickness) of the material.
 inline float computeEpsilon(float molarElectronDensity, float thickness,
-                            const RelativisticQuantities& rq) {
+                            const Acts::RelativisticQuantities& rq) {
   return 0.5f * K * molarElectronDensity * thickness * rq.q2OverBeta2;
 }
 
 /// Compute epsilon logarithmic derivative w/ respect to q/p.
-inline float logDeriveEpsilon(float qOverP, const RelativisticQuantities& rq) {
+inline float logDeriveEpsilon(float qOverP,
+                              const Acts::RelativisticQuantities& rq) {
   // only need to compute d(q²/beta²)/(q²/beta²); everything else cancels.
   return 2 / (qOverP * rq.gamma * rq.gamma);
 }
 
 /// Compute the density correction factor delta/2.
-///
-/// Uses RPP2018 eq. 33.6 which is only valid for high energies.
-///
-/// @todo Should we use RPP2018 eq. 33.7 instead w/ tabulated constants?
-inline float computeDeltaHalf(float meanExitationPotential,
+inline float computeDeltaHalf(const Acts::MaterialSlab& slab,
+                              float meanExitationPotential,
                               float molarElectronDensity,
-                              const RelativisticQuantities& rq) {
-  // only relevant for very high ernergies; use arbitrary cutoff
-  if (rq.betaGamma < 10.0f) {
-    return 0.0f;
+                              const Acts::RelativisticQuantities& rq) {
+  const auto density = slab.material().densityEffectData();
+  /// Uses RPP2018 eq. 33.6 which is only valid for high energies.
+  if (density == Acts::detail::DensityEffectData{}) {
+    // only relevant for very high ernergies; use arbitrary cutoff
+    if (rq.betaGamma < 10.0f) {
+      return 0.0f;
+    }
+    // pre-factor according to RPP2019 table 33.1
+    const auto plasmaEnergy =
+        PlasmaEnergyScale * std::sqrt(1000.f * molarElectronDensity);
```

Comment:
Wow.. The original Equation for density effect correction was NOT bad! It was just missing 1000 factor in calculating `plasmaEnergy`. It can reproduce the PDG value very well.

If people don't disagree I will delete density effect data in Material.

Reply:
But don't we have to store the density of the material anyways?

```
  parameters[eMolarDensity] = m_molarRho;
```

Reply:
> I will delete density effect data in Material.

You are not seeing that anymore in the PR because I already deleted it.

Reply:
There was sth like this in the PR but I removed it today:
https://github.com/acts-project/detray/blob/main/core/include/detray/materials/detail/density_effect_data.hpp